apps/core/src/modules/email-validator.ts

```diff
@@ -43,3 +44,52 @@ export async function getStudentData(ra: string) {
 
   return response;
 }
+
+export async function validateUserData(
+  emailToCheck: string,
+  ra: string,
+  app: FastifyInstance,
+) {
+  const checkUser = await getStudentData(ra);
+
+  if (!checkUser) {
+    throw new Error('RA_NOT_FOUND');
+  }
+
+  const emailList = Array.isArray(checkUser?.email) ? checkUser.email : [];
+  const employeePromises = emailList.map(
+    async (email) => await getEmployeeData(email),
+  );
+  const employees = await Promise.all(employeePromises);
+  const validEmployees = employees.filter((employee) => employee !== null);
+
+  if (validEmployees.length > 0) {
+    app.log.warn('UFABC employee', validEmployees);
+    throw new Error('HAS_UFABC_CONTRACT');
+  }
+
+  let email = '';
+
+  if (emailList.length === 0) {
+    app.log.warn({
+      ra,
+      username: checkUser.username,
+      msg: 'No email found, using username as email',
+    });
+    email = checkUser.username.concat('@aluno.ufabc.edu.br');
+  }
+
+  if (emailList.length > 0) {
+    email = emailList[0];
+  }
+
+  if (emailToCheck !== email) {
+    app.log.warn(
+      'Tentativa de burlar o email com proxy:',
+      emailToCheck,
+      '!==',
+      email,
+    );
```

Comment:
Aqui alguns pontos.
-Em cenarios de multipos parametros os logs devem ser feitos utilizando objetos, assim facilita a visualização.
- Logs em portugues nao devem ser utilizados, se ver em outro lugar no código pode remover
- A condicional ja te mostra que os emails sao diferentes
- Nao podemos afirmar que teve uso de proxy
```suggestion
    app.log.warn({emailToCheck, email}, 'wrong email received');
```